src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -49,6 +52,10 @@ public override T[] Rent(int minimumLength)
                     // If the bucket has an array left and returned it, give it to the caller
                     if (buffer != null)
                     {
+                        if (log.IsEnabled())
+                        {
+                            log.BufferRented(Utilities.GetBufferId<T>(buffer), buffer.Length, Utilities.GetBucketId<T>(_buckets[i]), Utilities.GetPoolId<T>(this));
```

Comment:
Nit: up to you, but I believe you can leave off the ```<T>``` and just let the compiler infer it in both Get*Id calls.